jfix-armeria-aggregating-profiler/src/test/kotlin/ru/fix/armeria/aggregating/profiler/ProfiledHttpClientTest.kt

```diff
@@ -451,6 +455,80 @@ internal class ProfiledHttpClientTest {
         }
     }
 
+    @Test
+    suspend fun `WHEN query string contains query parameters THEN they removed from metric 'path' label`() {
+        val profiler = AggregatingProfiler()
+        val profilerReporter = profiler.createReporter()
+        val path = "/pathWithQueryParams"
+        val mockServer = ArmeriaMockServer {
+            annotatedService(object : Any() {
+                @Get("/pathWithQueryParams")
+                fun testPath(queryParams: QueryParams): HttpResponse {
+                    logger.info { "GET Request with query params arrived: $queryParams" }
+                    return HttpResponse.of(HttpStatus.OK)
+                }
+            })
+        }
+        mockServer.start()
+        try {
+            val client = WebClient
+                .builder(mockServer.httpUri())
+                .decorator(ProfiledHttpClient.newDecorator(profiler))
+                .build()
+            try {
+                client.get("$path?testParam1=val1&testParam2=val2").aggregate().await()
+            } catch (e: Exception) {
+                logger.error(e)
+            }
+
+            val pathLabelOfConnectMetric: String = eventually(1.seconds) {
+                val report = profilerReporter.buildReportAndReset { metric, _ ->
+                    metric.name == Metrics.HTTP_CONNECT
+                }
+                val connectMetric = report.profiledCallReportWithNameEnding(Metrics.HTTP_CONNECT)
+                connectMetric should {
+                    it.shouldNotBeNull()
+                    it.stopSum shouldBe 1
+                    it.identity.tags shouldContainKey MetricTags.PATH
+                }
+                connectMetric!!.identity.tags[MetricTags.PATH]!!
+            }
+            val pathLabelOfConnectedMetric: String = eventually(1.seconds) {
+                val report = profilerReporter.buildReportAndReset { metric, _ ->
+                    metric.name == Metrics.HTTP_CONNECTED
+                }
+                val connectedMetric = report.profiledCallReportWithNameEnding(Metrics.HTTP_CONNECTED)
+                connectedMetric should {
+                    it.shouldNotBeNull()
+                    it.stopSum shouldBe 1
+                    it.identity.tags shouldContainKey MetricTags.PATH
+                }
+                connectedMetric!!.identity.tags[MetricTags.PATH]!!
+            }
+            val pathLabelOfHttpSuccessMetric: String = eventually(1.seconds) {
+                val report = profilerReporter.buildReportAndReset { metric, _ ->
+                    metric.name == Metrics.HTTP_SUCCESS
+                }
+                val httpSuccessMetric = report.profiledCallReportWithNameEnding(Metrics.HTTP_SUCCESS)
+                httpSuccessMetric should {
+                    it.shouldNotBeNull()
+
+                    it.identity.tags shouldContainKey MetricTags.PATH
+                }
+                httpSuccessMetric!!.identity.tags[MetricTags.PATH]!!
+            }
```

Comment:
It seems like we can create and use one new parametrized assertion method instead of three very similiar parts of code